lib/request.js

```diff
@@ -23,12 +23,14 @@ var doRequest = (url, qs, cb = (() => false)) => {
       cb(err, res)
     }
 
-    return http({
+    Object.assign(http_params, {
```

Comment:
I think we are generating side effects here since we are mutation the input object by reference, which is not good. Just merge the object into a new one without side effects.